waspls/src/Wasp/LSP/State.hs

```diff
@@ -0,0 +1,46 @@
+module Wasp.LSP.State
```

Comment:
Taking into accoutn this file is named State, it seems you put too much stuff into it? You are using it more like Types? I guess that is ok because we are still trying to figure out what we need, but ideally we would organize things a bit better I think?

Reply:
I agree, there ended up being a lot of unrelated stuff in here! I think since most of it is essentially boilerplate right now (since we have no state or server config right now), maybe we leave it in this sort of messy state until stuff gets added here?

Reply:
I would still try to clean it up immediately, just because later it might get stuck like this for long time :D. We can try splitting stuff into multiple modules, or we can rename module to Common.

Reply:
Renamed to Core.